Training/strings.cpp

```diff
@@ -7,7 +7,10 @@ using namespace std;
 int hackerrank::strings()
 {
     string a, b;
+    //ES- Note: Cin on strings won't capture whitespaces.
+    //Consider checking out getline.
```

Comment:
No, not really. Just a consideration. When you get more into console I/O, it can sometimes be preferably to eat an entire line so that you can prune the input buffer better. IE. if you `int a; cin>> a;` and the user doesn't type an integer in, it doesn't take the input off of the buffer, causing all subsequent `cin` calls to misbehave.

Reply:
Would using `fgets` then be preferable to `getline` or pretty much the same?

Reply:
Uhh... either really. That's more of a C vs. C++ consideration. If you're using one set of functions you should stick with those typically.

In general, the handling of stream concepts in the C++ standard library isn't a thing people are super happy about, but there are some rather complex things that you can express very elegantly using it. I'd recommend learning both really, just avoid mix and matching them _too_ much. Certainly not in the same block of code and try to ensure that you leave the I/O buffers in healthy shape.

You can even use `scanf` with regex to grab a full-line, but `fgets` would be preferred in such a case.

Oddly, I have tended toward C++ style __file__ IO in the past, but prefer C style IO for __console__. I think if I were working on any IO now-a-days and I cared about run-time performance of streaming data, I'd use C style or a third party library.

Reply:
Understood.